src/library_fs.js

```diff
@@ -1271,13 +1271,13 @@ mergeInto(LibraryManager.library, {
       FS.mkdev('/dev/tty1', FS.makedev(6, 0));
       // setup /dev/[u]random
       var random_device;
-      if (typeof crypto !== 'undefined') {
+      if (typeof crypto !== 'undefined' && crypto['getRandomValues'] !== 'function') {
         // for modern web browsers
         var randomBuffer = new Uint8Array(1);
         random_device = function() { crypto.getRandomValues(randomBuffer); return randomBuffer[0]; };
       } else if (ENVIRONMENT_IS_NODE) {
-        // for nodejs
-        random_device = function() { return require('crypto')['randomBytes'](1)[0]; };
+        // for nodejs with or without crypto support included
+        random_device = function() { try { return require('crypto')['randomBytes'](1)[0]; } catch(e) { return (Math.random()*256)|0; }; };
```

Comment:
Ok, aside from the wrong condition discussed above, the last thing I think is to clean up this line. First, let's not have it all on one line. Second, please check if crypto is available once, and set `random_device` to the right value based on that, so it doesn't do the try-catch check every time it's called.